exovetter/main.py

```diff
@@ -0,0 +1,61 @@
+
+#A sketch of an architecture. For discussion
+
+#Each vetting test is wrapped in a class that follows this
+#structure.
+class Vetter():
+    def __init__(self, **kwargs):
+        """kwargs stores the configuration parameters common to all TCEs
+        For example, for the Odd-even test, it might specify the signficance
+        of the depth difference that causes a TCE to fail
+        """
+        pass
+
+    def apply(self, tce, lightcurve):
+        """Actually run the test. Returns a dictionary of metric values"""
+        pass
+
+    def plot(self, tce, lightcurve):
+        """Optional, generate a diagnostic plot"""
+        pass
+
+
+#The LPP vetter is an example of a Vetter class.
+class Lpp(Vetter):
+    ...
+
+    def apply(self, tce, lightcurve):
+        #Actual implementation of LPP is called here
+        ...
+
+#The odd even test. We can have many such tests
+class OddEven(Vetter):
+    pass
+
+
+def main(config):
+    tceTable = pd.read_csv('tcelist.csv')
```

Comment:
Camel-case should not be used but we can table that for when we move on to style discussions. 😉